pkg/storage/gcp/saver.go

```diff
@@ -0,0 +1,102 @@
+package gcp
+
+import (
+	"fmt"
+	"strings"
+
+	"cloud.google.com/go/storage"
+	"github.com/bketelsen/buffet"
+	"github.com/gobuffalo/buffalo"
+	"google.golang.org/api/option"
+)
+
+// Storage implements the (github.com/gomods/pkg/storage).Saver interface
+type Storage struct {
+	bucket *storage.BucketHandle
+}
+
+// New returns a new Storage instance
+// authenticated using the provided ClientOptions
+func New(ctx buffalo.Context, projectID, bucket string, cred option.ClientOption) (*Storage, error) {
+	client, err := storage.NewClient(ctx, cred)
+	if err != nil {
+		return nil, fmt.Errorf("could not create new client: %s", err)
+	}
+	// TODO: client should be closed on any error or when done
+	// should this be done after save is complete?
```

Comment:
@robjloranger you could provide a `Close` method on the `Storage` struct and then require that the caller call it after they're done with the saver. I think in most cases, they'll never want to close it (because in most cases there's gonna be a goroutine filling caches, etc...) but it'll be good to have

Reply:
I was just looking at the docs for the library and they mention the close does not need to be closed at program exit. So maybe we don't really need it.

Reply:
@arschles do you still think we need a close method here?